src/app/app.js

```diff
@@ -1,21 +1,31 @@
 import { CssVariables } from '@dhis2/ui'
 import React from 'react'
+import { HashRouter as Router, Route } from 'react-router-dom'
+import { QueryParamProvider } from 'use-query-params'
 import './app.css'
+import { ContextSelection } from '../context-selection/index.js'
 import { Layout } from './layout/index.js'
 
+/**
+ * "use-query-params" requires a router. It suggests react-router-dom in the
+ * docs, so that's why I chose that one. Unfortunately it doesn't work with
+ * react-router-dom@6, so I have do go with react-router-dom@^5
+ */
```

Comment:
You could go with `reach-router` instead, `use-query-params` is compatible with that as well. From the reach docs:

> Reach Router and it’s sibling project React Router are merging as React Router v6. In other words, Reach Router v2 and React Router v6 are the same. There is more information on the maintainers website.
>
> https://reach.tech/router/

So if you can't use the latest react-router due to compatibility issues, I'd look into reach-router, of which you should be to use the latest version. Might result in us having to make less changes later.

Reply:
These projects merged and are being released as react-router v6, which is exactly the one that didn't work